Makes stepladder wearable.

JonahDes opened the pull request:
```SUMMARY: Balance "Allow stepladders to be wearable."```
#### Summary

A bindle is often held with one hand and still enables use of the other, it is classed as an armor item; by the same reasoning, a stepladder can be held over a shoulder and should still allow hand and arm usage, at an extraordinary cost to encumbrance value.

After testing it was assumed that id conflicts stopped this being possible, so stepladder was moved to tools_armor.json

Images in-game. Works.

https://media.discordapp.net/attachments/365907426270117888/579373243509768205/unknown.png

![image](https://user-images.githubusercontent.com/50489005/57973583-f09d1500-79a2-11e9-93c7-0354fd56b70c.png)

DracoGriffin commented:
Hello and welcome @JonahDes!

Thanks for the addition and make sure to follow the template provided when you make a PR post, specifically this component:

```
#### Summary
<!--
A one-line description of your change that will be extracted and added to the [project changelog](https://github.com/CleverRaven/Cataclysm-DDA/blob/master/data/changelog.txt).

The format is (ignore the square brackets): ```SUMMARY: [Category] "[description]"```

The categories to choose from are:

* Features
* Content
* Interface
* Mods
* Balance
* Bugfixes
* Performance
* Infrastructure
* Build
* I18N

Example: ```SUMMARY: Content "Adds new mutation category 'Mouse'"```

See the [Changelog Guidelines](https://github.com/CleverRaven/Cataclysm-DDA/blob/master/doc/CHANGELOG_GUIDELINES.md) for explanations of the categories.
-->
```

So based on your change, you should edit your post to have something like this:

    ```SUMMARY: Balance "Allow stepladders to be wearable."```

If you have any further questions, feel free to reply to your post or ask on Discord (#mod-and-dev-general)/Discourse.

JonahDes commented:
> Hello and welcome @JonahDes!
> 
> Thanks for the addition and make sure to follow the template provided when you make a PR post, specifically this component:
> 
> ```
> #### Summary
> <!--
> A one-line description of your change that will be extracted and added to the [project changelog](https://github.com/CleverRaven/Cataclysm-DDA/blob/master/data/changelog.txt).
> 
> The format is (ignore the square brackets): ```SUMMARY: [Category] "[description]"```
> 
> The categories to choose from are:
> 
> * Features
> * Content
> * Interface
> * Mods
> * Balance
> * Bugfixes
> * Performance
> * Infrastructure
> * Build
> * I18N
> 
> Example: ```SUMMARY: Content "Adds new mutation category 'Mouse'"```
> 
> See the [Changelog Guidelines](https://github.com/CleverRaven/Cataclysm-DDA/blob/master/doc/CHANGELOG_GUIDELINES.md) for explanations of the categories.
> -->
> ```
> 
> So based on your change, you should edit your post to have something like this:
> 
> ```
> ```SUMMARY: Balance "Allow stepladders to be wearable."```
> ```
> 
> If you have any further questions, feel free to reply to your post or ask on Discord (#mod-and-dev-general)/Discourse.

Added some details to original post.

ZhilkinSerg commented:
Doesn't work - stepladder is not wearable.

Rail-Runner commented on data/json/items/tool_armor.json:
```diff
@@ -1271,6 +1271,26 @@
+    "encumbrance": 120,
+    "material_thickness": 1,
+    "use_action": "LADDER",
+    "flags": [ "OVERSIZE", "RESTRICT_HANDS", "WATER_FRIENDLY" ]
```
Needs "BELTED" flag here too, so that the stepladder is layered over all other worn items.

Rail-Runner commented:
I'm pretty sure the problem is that the old stepladder item entry wasn't removed. If it's removed, it should work just fine.

JonahDes commented:
> Doesn't work - stepladder is not wearable.

Yeah, I'll try removing the entries from other JSONs, merging it into the PR and seeing how that works.

kevingranade commented:
Your rationale is incorrect.  The only reason a bindle is wearable is to allow it to contribute to inventory capacity.  This hack should not be extended any further.

JonahDes commented:
> Your rationale is incorrect. The only reason a bindle is wearable is to allow it to contribute to inventory capacity. This hack should not be extended any further.

Alright I'll make rationale which matches the bindle, then. By making the ladder wearable, you can carry more.

Rail-Runner commented:
> > Your rationale is incorrect. The only reason a bindle is wearable is to allow it to contribute to inventory capacity. This hack should not be extended any further.
> 
> Alright I'll make rationale which matches the bindle, then. By making the ladder wearable, you can carry more.

But being able to carry more if you're holding a ladder doesn't make any sense.

JonahDes commented:
> > > Your rationale is incorrect. The only reason a bindle is wearable is to allow it to contribute to inventory capacity. This hack should not be extended any further.
> > 
> > 
> > Alright I'll make rationale which matches the bindle, then. By making the ladder wearable, you can carry more.
> 
> But being able to carry more if you're holding a ladder doesn't make any sense.

Before wearable; you'd have to choose between wielding the ladder or not having it for a weapon.
After wearable; you'd be able to effectively wear the ladder, whilst not having to leave your weapon behind.

Hence, increased carrying capacity.

Rail-Runner commented:
> Before wearable; you'd have to choose between wielding the ladder or not having it for a weapon.
> After wearable; you'd be able to effectively wear the ladder, whilst not having to leave your weapon behind.
> 
> Hence, increased carrying capacity.

Got it. Should have been "free up some storage volume to carry it in a hand" then, rather than just carrying more.
